src/node/historical_queries_adapter.cpp

```diff
@@ -298,7 +298,7 @@ namespace ccf::historical
               hpubkey,
               ReplicatedNetworkIdentity::subject_name,
               {},
-              ds::to_x509_time_string(crypto::OpenSSL::to_time_t(NULL)),
+              ds::to_x509_time_string(std::chrono::system_clock::now()),
```

Comment:
What does this return in an SGX enclave?

Reply:
The current time, which I believe it gets from the host. I haven't checked that, but I figured if `time(NULL)` works, this should too. I'll take a closer look later today.

Reply:
https://github.com/microsoft/CCF/issues/2468 may give you some more context on OCALLs and time. Maybe we could simply extract the validity period from the historical certificate and use it in the newly-endorsed certificate?

Reply:
Yeah, that's a good idea, we really shouldn't depend on the host time anywhere. I would rather take the times from the node cert though, because that's the endorser.